lib/domain/usecase/room/chat_get_pinned_events_interactor.dart

```diff
@@ -9,12 +9,17 @@ import 'package:matrix/matrix.dart';
 class ChatGetPinnedEventsInteractor {
   Stream<Either<Failure, Success>> execute({
     required Room room,
+    List<Event?>? eventsInMem,
   }) async* {
     Logs().d(
       "ChatGetPinnedEventsInteractor()::execute()::roomId: ${room.id}",
     );
     yield Right(ChatGetPinnedEventsLoading());
     try {
+      if (eventsInMem != null && eventsInMem.isNotEmpty) {
+        yield Right(ChatGetPinnedEventsSuccess(pinnedEvents: eventsInMem));
```

Comment:
`eventsInMem` is empty not `ChatGetPinnedEventsSuccess`. It doesn't make sense, let's create another more specific state

Reply:
Why handle it inside controller?

Reply:
removed